explainability/core/counterfactuals/src/main/java/org/kie/kogito/explainability/local/counterfactual/CounterfactualExplainer.java

```diff
@@ -105,10 +114,22 @@ private Executor getExecutor() {
 
         for (int i = 0; i < predictionInput.getFeatures().size(); i++) {
             final Feature feature = predictionInput.getFeatures().get(i);
-            final Boolean isConstrained = constraints.get(i);
-            final FeatureBoundary featureDistribution = dataBoundaries.getFeatureBoundaries().get(i);
-            final CounterfactualEntity counterfactualEntity = CounterfactualEntityFactory.from(feature, isConstrained, featureDistribution);
-            entities.add(counterfactualEntity);
+
+            if (feature.getType() == Type.CATEGORICAL) {
+                final Boolean isConstrained = constraints.get(i);
+                final Set<String> categories = this.categoriesMap.get(feature.getName());
+                if (categories==null || categories.isEmpty()) {
+                    logger.warn("Categorical entity {} has no associated categories", feature.getName());
+                }
+                CounterfactualEntity counterfactualEntity = CategoricalEntity.from(feature, categories, isConstrained);
+                entities.add(counterfactualEntity);
+            } else {
+
+                final Boolean isConstrained = constraints.get(i);
+                final FeatureBoundary featureDistribution = dataBoundaries.getFeatureBoundaries().get(i);
+                final CounterfactualEntity counterfactualEntity = CounterfactualEntityFactory.from(feature, isConstrained, featureDistribution);
+                entities.add(counterfactualEntity);
+            }
```

Comment:
@danielezonca very good point!
I've changed `FeatureBoundary` to `FeatureDomain` so that now it can include the associated values for categorical features.
The `CounterfactualEntityFactory` now works with this generic `FeatureDomain`.
Also, this means that it's not needed to pass a separate map with the values to the explainer anymore.